app/src/main/java/org/mozilla/focus/navigation/MainActivityNavigation.kt

```diff
@@ -135,20 +138,37 @@ class MainActivityNavigation(
     }
 
     /**
-     * Show first run onboarding.
+     * Show first run onBoarding.
      */
     fun firstRun() {
         val onboardingFeature = FocusNimbus.features.onboarding
         val onboardingConfig = onboardingFeature.value(activity)
-        val onboardingFragment = if (onboardingConfig.isEnabled) {
+
+        val onBoardingFragment = if (onboardingConfig.isEnabled) {
             onboardingFeature.recordExposure()
-            OnboardingFragment.create()
+            val onBoardingStorage = OnBoardingStorage(activity)
+            when (onBoardingStorage.getCurrentOnBoardingStep()) {
+                OnBoardingStep.ONBOARDINGFIRSTSCREEN -> {
+                    OnBoardingFirstFragment.create()
+                }
+                OnBoardingStep.ONBOARDINGSECONDSCREEN -> {
+                    OnBoardingSecondFragment.create()
+                }
+            }
         } else {
             FirstrunFragment.create()
         }
+
+        activity.supportFragmentManager
+            .beginTransaction()
+            .replace(R.id.container, onBoardingFragment, FirstrunFragment.FRAGMENT_TAG)
```

Comment:
Not sure if this has an impact now but seems safer to use the tag for the fragment that's actually shown instead of always using the tag from `FirstrunFragment`.

Reply:
Done